Validate total service DNS hostname+domain length

SpComb opened the pull request:
Docker imposes a limit of 64 chars on the container FQDN (combined `hostname.domain`), because docker/runc will call `sethostname` with the full FQDN. This results in a rather useless Docker API error if the FQDN is too long.

Fixes #825 by failing the service create with a validation error, instead of failing on the service container deploy.

The logic is slightly more complex because it also needs to take into account the number of digits required to represent the container instance number.

## Example

```yaml
stack: test/test-asdfasdfasdfasdf
services:
  xxxxxxxx10xx12:
    image: redis

  xxxxxxxx10xxx13:
    image: redis
    depends_on:
      - xxxxxxxx10xx12
```

### Before
#### `kontena stack install test/test-longname.yaml`
```
 [done] Creating stack test-asdfasdfasdfasdf      
 [done] Triggering deployment of stack test-asdfasdfasdfasdf     
 [done] Waiting for deployment to start     
 [done] Deploying service xxxxxxxx10xx12     
 [fail] Deploying service xxxxxxxx10xxx13     
Deployment of service development/test-asdfasdfasdfasdf/xxxxxxxx10xxx13 failed:
  - halting deploy of development/test-asdfasdfasdfasdf/xxxxxxxx10xxx13, one or more instances failed
  - GridServiceInstanceDeployer::ServiceError: Docker::Error::ServerError: invalid header field value "oci runtime error: container_linux.go:247: starting container process caused \"process_linux.go:359: container init caused \\\"invalid argument\\\"\"\n" (on node core-01)
```

### After
#### `kontena stack install test/test-longname.yaml`
```
 [fail] Creating stack test-asdfasdfasdfasdf      
 [error] 422 : Unprocessable Entity:
	services:
	  xxxxxxxx10xxx13:
	    name: 'Total grid service name length 65 is over limit (64): xxxxxxxx10xxx13-1.test-asdfasdfasdfasdf.development.kontena.local'
```

jnummelin commented on docs/references/kontena-yml.md:
```diff
@@ -52,17 +52,28 @@ services:
 
+The service name must start with a letter, number or underscore, and may also contain dashes.
+
+There is no limitation on the length of a service name per se, but the resulting service container FQDN (`service-1.stack-name.grid-name.kontena.local`) must be 64 bytes or less. The effective service name length maximum thus also depends on the grid and stack names, so they should be kept reasonably short.
```
Maybe add note that it's really docker/runc restriction

SpComb replied:
There isn't really any issue to link to for it, though... probably need to create one first?

SpComb commented on docs/using-kontena/grids.md:
```diff
@@ -11,10 +11,17 @@ The [Grid](../core-concepts/architecture.md#the-grid) is a top-level object in K
+
+### Grid name
+
+The grid name will be used as part of the `grid-name.kontena.local` DNS domain used for service lookups. The grid name may consist of letters (`a-z`), digits (`0-9`), underscores (`_`) and dashes (`-`).
```
BTW I wonder how well those `_` would work... or any `-` at the start?

SpComb commented:
Installing a stack with two services, one under the name limit, the other over, results in an installed stack with missing services... the `Stacks::Create#validate` -> `validate_services` calls `GridServices::Create#validate` with the service definition, but *without* the `stack`, so the service name validation doesn't take the length of the stack name into account, and then it fails during the `create_services` :(

Not sure if worth fixing...

```
$ bin/kontena stack install test/test-longname.yaml 
 [fail] Creating stack test-asdfasdfasdfasdf      
 [error] 422 : Unprocessable Entity:
	services:
	  xxxxxxxx10xxx13:
	    name: 'Total grid service name length 65 is over limit (64): xxxxxxxx10xxx13-1.test-asdfasdfasdfasdf.development.kontena.local'
$ bin/kontena stack show test-asdfasdfasdfasdf
test-asdfasdfasdfasdf:
  created: 2017-09-19T08:47:24.964Z
  updated: 2017-09-19T08:47:24.966Z
  state: initialized
  stack: test/test-asdfasdfasdfasdf
  version: 0.0.1
  revision: 1
  expose: -
  parent: -
  services:
    xxxxxxxx10xx12:
      created: 2017-09-19T08:47:24.976Z
      updated: 2017-09-19T08:47:24.976Z
      image: redis:latest
      revision: 
      state: initialized
      stateful: no
      scaling: 1
      strategy: ha
      read_only: no
      deploy_opts:
        min_health: 
      dns: xxxxxxxx10xx12.test-asdfasdfasdfasdf.development.kontena.local
```
